{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "**Description of the problem**\r\n\r\nInstall issues on Mac OSX using Miniconda (not anaconda) distribution.\r\nI followed the conda install instructions, added $HOME/miniconda3/lib to GMT_LIBRARY_PATH and still had issues. \r\ntbb comes automatically with the Anaconda distribution but not the Miniconda distribution, therefore libtbb could not be found from libtiledb.\r\n\r\nThe solution was to add \"tbb\" to the list of packages before installing gmt, and then pygmt.\r\ni.e. numpy pandas xarray netcdf4 packaging tbb gmt\r\n\r\nIt's pretty minor, but it might be worth adding tbb to the install instructions.\r\nCheers!\r\n\r\n**Full code that generated the error**\r\n\r\n```python\r\nimport pygmt\r\n```\r\n\r\n**Full error message**\r\n\r\n```\r\ndyld: Library not loaded: @rpath/libtbb.dylib\r\n  Referenced from: /Users/nfoks/miniconda3/lib/libtiledb.dylib\r\n  Reason: image not found\r\nTraceback (most recent call last):\r\n  File \"<stdin>\", line 1, in <module>\r\n  File \"/Users/nfoks/miniconda3/lib/python3.8/site-packages/pygmt/__init__.py\", line 51, in <module>\r\n    _begin()\r\n  File \"/Users/nfoks/miniconda3/lib/python3.8/site-packages/pygmt/session_management.py\", line 16, in begin\r\n    with Session() as lib:\r\n  File \"/Users/nfoks/miniconda3/lib/python3.8/site-packages/pygmt/clib/session.py\", line 184, in __enter__\r\n    self.create(\"pygmt-session\")\r\n  File \"/Users/nfoks/miniconda3/lib/python3.8/site-packages/pygmt/clib/session.py\", line 331, in create\r\n    c_create_session = self.get_libgmt_func(\r\n  File \"/Users/nfoks/miniconda3/lib/python3.8/site-packages/pygmt/clib/session.py\", line 283, in get_libgmt_func\r\n    self._libgmt = load_libgmt()\r\n  File \"/Users/nfoks/miniconda3/lib/python3.8/site-packages/pygmt/clib/loading.py\", line 59, in load_libgmt\r\n    raise GMTCLibNotFoundError(\"\\n\".join(error_msg))\r\npygmt.exceptions.GMTCLibNotFoundError: Error loading GMT shared library at 'libgmt.dylib'.\r\ndlopen(libgmt.dylib, 6): Library not loaded: @rpath/libtbb.dylib\r\n  Referenced from: /Users/nfoks/miniconda3/lib/libtiledb.dylib\r\n  Reason: image not found\r\n```\r\n\r\n**System information**\r\n\r\nPlease paste the output of `python -c \"import pygmt; pygmt.show_versions()\"`:\r\n\r\n```\r\nPyGMT information:\r\n  version: v0.3.1\r\nSystem information:\r\n  python: 3.8.5 (default, Sep  4 2020, 02:22:02)  [Clang 10.0.0 ]\r\n  executable: /Users/nfoks/miniconda3/bin/python\r\n  machine: macOS-10.14.6-x86_64-i386-64bit\r\nDependency information:\r\n  numpy: 1.20.3\r\n  pandas: 1.0.5\r\n  xarray: 0.18.2\r\n  netCDF4: 1.5.6\r\n  packaging: 20.9\r\n  ghostscript: 9.54.0\r\n  gmt: 6.1.1\r\nGMT library information:\r\n  binary dir: /Users/nfoks/miniconda3/bin\r\n  cores: 8\r\n  grid layout: rows\r\n  library path: /Users/nfoks/miniconda3/lib/libgmt.dylib\r\n  padding: 2\r\n  plugin dir: /Users/nfoks/miniconda3/lib/gmt/plugins\r\n  share dir: /Users/nfoks/miniconda3/share/gmt\r\n  version: 6.1.1\r\n```\r\n",
    "closed_at": "2021-05-25T22:52:26Z",
    "closed_by": {
        "avatar_url": "https://avatars.githubusercontent.com/u/15173535?v=4",
        "events_url": "https://api.github.com/users/leonfoks/events{/privacy}",
        "followers_url": "https://api.github.com/users/leonfoks/followers",
        "following_url": "https://api.github.com/users/leonfoks/following{/other_user}",
        "gists_url": "https://api.github.com/users/leonfoks/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/leonfoks",
        "id": 15173535,
        "login": "leonfoks",
        "node_id": "MDQ6VXNlcjE1MTczNTM1",
        "organizations_url": "https://api.github.com/users/leonfoks/orgs",
        "received_events_url": "https://api.github.com/users/leonfoks/received_events",
        "repos_url": "https://api.github.com/users/leonfoks/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/leonfoks/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/leonfoks/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/leonfoks"
    },
    "comments": 4,
    "comments_content": [
        {
            "author_association": "NONE",
            "body": "\ud83d\udc4b Thanks for opening your first issue here! Please make sure you filled out the template with as much detail as possible. You might also want to take a look at our [contributing guidelines](https://github.com/GenericMappingTools/pygmt/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/GenericMappingTools/pygmt/blob/master/CODE_OF_CONDUCT.md).\n",
            "created_at": "2021-05-25T20:19:40Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1294#issuecomment-848232263",
            "id": 848232263,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1294",
            "node_id": "MDEyOklzc3VlQ29tbWVudDg0ODIzMjI2Mw==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/848232263/reactions"
            },
            "updated_at": "2021-05-25T20:19:40Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/848232263",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/in/4141?v=4",
                "events_url": "https://api.github.com/users/welcome%5Bbot%5D/events{/privacy}",
                "followers_url": "https://api.github.com/users/welcome%5Bbot%5D/followers",
                "following_url": "https://api.github.com/users/welcome%5Bbot%5D/following{/other_user}",
                "gists_url": "https://api.github.com/users/welcome%5Bbot%5D/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/apps/welcome",
                "id": 30606887,
                "login": "welcome[bot]",
                "node_id": "MDM6Qm90MzA2MDY4ODc=",
                "organizations_url": "https://api.github.com/users/welcome%5Bbot%5D/orgs",
                "received_events_url": "https://api.github.com/users/welcome%5Bbot%5D/received_events",
                "repos_url": "https://api.github.com/users/welcome%5Bbot%5D/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/welcome%5Bbot%5D/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/welcome%5Bbot%5D/subscriptions",
                "type": "Bot",
                "url": "https://api.github.com/users/welcome%5Bbot%5D"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "HI @leonfoks, could you provide the full output from running `conda list`? I'm wondering if there's another dependency interfering with the install on miniconda. Two possible issues I spotted were:\r\n\r\n1. The installed `pandas` version (1.0.5) is about a year old, the most recent one is 1.2.4 (which is why I suspect there could be a package conflict somewhere).\r\n2. Adding `$HOME/miniconda3/lib` to `GMT_LIBRARY_PATH` shouldn't be needed if you did `conda activate pygmt`.\r\n\r\nWe'll check if other macOS users can reproduce this too. Cc @seisman if you have time.",
            "created_at": "2021-05-25T22:34:55Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1294#issuecomment-848313370",
            "id": 848313370,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1294",
            "node_id": "MDEyOklzc3VlQ29tbWVudDg0ODMxMzM3MA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/848313370/reactions"
            },
            "updated_at": "2021-05-25T22:34:55Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/848313370",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        },
        {
            "author_association": "NONE",
            "body": "Aha! that worked. I think my folly was forgetting to activate the environment. Using the pygmt env installed Pandas 1.2.4 and it worked fine! Sorry about that and thanks for getting back so quickly.",
            "created_at": "2021-05-25T22:52:26Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1294#issuecomment-848320319",
            "id": 848320319,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1294",
            "node_id": "MDEyOklzc3VlQ29tbWVudDg0ODMyMDMxOQ==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/848320319/reactions"
            },
            "updated_at": "2021-05-25T22:52:26Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/848320319",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/15173535?v=4",
                "events_url": "https://api.github.com/users/leonfoks/events{/privacy}",
                "followers_url": "https://api.github.com/users/leonfoks/followers",
                "following_url": "https://api.github.com/users/leonfoks/following{/other_user}",
                "gists_url": "https://api.github.com/users/leonfoks/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/leonfoks",
                "id": 15173535,
                "login": "leonfoks",
                "node_id": "MDQ6VXNlcjE1MTczNTM1",
                "organizations_url": "https://api.github.com/users/leonfoks/orgs",
                "received_events_url": "https://api.github.com/users/leonfoks/received_events",
                "repos_url": "https://api.github.com/users/leonfoks/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/leonfoks/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/leonfoks/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/leonfoks"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Glad to help out, and thanks for trying out PyGMT :grin:",
            "created_at": "2021-05-25T23:06:28Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1294#issuecomment-848328112",
            "id": 848328112,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1294",
            "node_id": "MDEyOklzc3VlQ29tbWVudDg0ODMyODExMg==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/848328112/reactions"
            },
            "updated_at": "2021-05-25T23:06:28Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/848328112",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        }
    ],
    "comments_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1294/comments",
    "created_at": "2021-05-25T20:19:38Z",
    "events_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1294/events",
    "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1294",
    "id": 901320131,
    "labels": [
        {
            "color": "ee0701",
            "default": true,
            "description": "Something isn't working",
            "id": 563309827,
            "name": "bug",
            "node_id": "MDU6TGFiZWw1NjMzMDk4Mjc=",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/labels/bug"
        }
    ],
    "labels_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1294/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "MDU6SXNzdWU5MDEzMjAxMzE=",
    "number": 1294,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1294/reactions"
    },
    "repository_url": "https://api.github.com/repos/GenericMappingTools/pygmt",
    "state": "closed",
    "state_reason": "completed",
    "timeline_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1294/timeline",
    "title": "Installation using Miniconda needs additional required package \"tbb\"",
    "updated_at": "2021-05-25T23:06:28Z",
    "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1294",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/15173535?v=4",
        "events_url": "https://api.github.com/users/leonfoks/events{/privacy}",
        "followers_url": "https://api.github.com/users/leonfoks/followers",
        "following_url": "https://api.github.com/users/leonfoks/following{/other_user}",
        "gists_url": "https://api.github.com/users/leonfoks/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/leonfoks",
        "id": 15173535,
        "login": "leonfoks",
        "node_id": "MDQ6VXNlcjE1MTczNTM1",
        "organizations_url": "https://api.github.com/users/leonfoks/orgs",
        "received_events_url": "https://api.github.com/users/leonfoks/received_events",
        "repos_url": "https://api.github.com/users/leonfoks/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/leonfoks/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/leonfoks/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/leonfoks"
    }
}